monasca_perf/alarm_perf.py

```diff
@@ -0,0 +1,142 @@
+import sys
+import time
+import multiprocessing
+
+from monascaclient import client
+from monascaclient import ksclient
+
+import warnings
+def no_warnings(message, category, filename, lineno):
+    pass
+warnings.showwarning = no_warnings
+
+num_processes = 10
+num_requests = 10
+num_metrics = 10
+
+# specify if the test should remove the generated alarms
+cleanup_after_test = False
+
+keystone = {
+        'username':'mini-mon',
+        'password':'password',
+        'project':'test',
+        'auth_url':'http://10.22.156.11:35357/v3'
+    }
+
+urls = [
+    'https://mon-ae1test-monasca01.useast.hpcloud.net:8080/v2.0',
+    'https://mon-ae1test-monasca02.useast.hpcloud.net:8080/v2.0',
+    'https://mon-ae1test-monasca03.useast.hpcloud.net:8080/v2.0',
+]
+
+metric_name = 'alarm_perf'
+metric_dimension = 'dim1'
+
+alarm_def_name = 'alarm_perf_test'
+alarm_def_expression = '{} > 0'.format(metric_name)
+
+def cleanup(monasca_client, name):
+    for definition in monasca_client.alarm_definitions.list():
+        if definition['name'] == name:
+            monasca_client.alarm_definitions.delete(**{'alarm_id':definition['id']})
+
+def create_alarm_definition(monasca_client):
+    try:
+        resp = monasca_client.alarm_definitions.create(**{
+            'name': alarm_def_name,
+            'expression': alarm_def_expression,
+            'match_by': [metric_dimension]
+        })
+        print('Alarm Definition ID: {}'.format(resp['id']))
+        return resp['id']
+    except Exception as ex:
+        print('Could not create alarm definition\n{}'.format(ex))
+        return None
+
+def do_work(url_queue, proc_id, token):
+    url = url_queue.get()
+    monasca_client = client.Client('2_0', url, **{'token':token})
+    try:
+        for i in xrange(num_requests):
```

Comment:
xrange disappears in python 3, not a big deal to keep it here but not a bad idea to change also.